### PR TITLE
QsLog and valgrind

### PR DESCRIPTION
**Pull request by Arne-W**
Here is my new PR. Picked the important changes from old PR #944 and added the redefinition of the QsLog macros. This should workaround the QsLog issue #941.
The other commits fix some valgrind issues, a memleak and a race condition.

### CODE REVIEW

File: src/ui/AP2DataPlot2D.h [hunk 1]
```diff
@@ -257,7 +257,7 @@ private slots:
     qint64 m_startIndex; //epoch msecs since graphing started
     QAction *m_addGraphAction;
     UASInterface *m_uas;
-    QSharedPointer<QProgressDialog> m_progressDialog;
```
Comment (billbonney) [inline src/ui/AP2DataPlot2D.h]: http://doc.qt.io/qt-5/qsharedpointer.html

QSahred pointer is always constructed as NULL and it deletes what is referenced when it goes out of scope. So the valugrind messages are not valid in this case.

**Comment by Arne-W** [comment]
The problem was not the a valgrind message about a memleak. The problem was the deletion in the threadDone method while not all emitted signals were processed. So the call `loadProgress` tried to access a deleted object. The progress dialogue is now deleted by the `deleteLater` method.

**Comment by billbonney** [comment]
@Arne-W Roger, that. The fix was to add deleteLater() No need to removed the QSharedPointer declaration or init it to 0. It's a shame that AP2 doesn't use more smart pointers. :)

**Comment by Arne-W** [comment]
I did not want create an issue because of the deleteLater and the delete which would be done by the smartpointer, as it could not guaranteed which comes first.
But you are right, it is possible to pass a deleter method to the pointer - cool feature - I will revert the smartpointer remove today.
And you are right - too much "old school" pointers. We will change them in the future :smiley:

**Comment by billbonney** [comment]
Merged with 4d0fc86f8e62c1923ec8897e4692f34070622d4e and a few previous commits. Thx :)

**Comment by Arne-W** [comment]
OK - I'll close it

**Pull request closed without merging**